### PR TITLE
[Fix-13972][dao] set default value for command

### LINKED ISSUE #13972: [Bug] [master] the task depend type is null when recovery suspend
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

image

image

When I trigger recovery suspend, the taskDependType of the Command in the database is set to null. But when the master node get the Command, it becomes Task_ only. The check about `checkCmdParam` failed.

### What you expected to happen

recovery suspend success

### How to reproduce

recovery suspend

### Anything else

I found `Command` have default `taskDependType`, but it is not useful with `@Builder`. Maybe should use `@Builder.Default`

### Version

dev

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by eye-gu**
## Purpose of the pull request

close #13972

## Brief change log

1. upgrade mysql version
2. make task depend type default

### CODE REVIEW

File: dolphinscheduler-bom/pom.xml [hunk 1]
```diff
@@ -56,7 +56,7 @@
         <hadoop.version>3.2.4</hadoop.version>
         <cron-utils.version>9.1.6</cron-utils.version>
         <h2.version>2.2.220</h2.version>
-        <mysql-connector.version>8.0.16</mysql-connector.version>
+        <mysql-connector.version>8.0.33</mysql-connector.version>
```
Comment (SbloodyS) [inline dolphinscheduler-bom/pom.xml]: This is a significant change for users. We should disscuss it. @ruanwenjun @zhongjiajie @EricGao888

Comment (zhongjiajie) [reply dolphinscheduler-bom/pom.xml]: it seem 8.0.33 is compatible with 8.0.16, But I want to know is why should we upgrade the version of mysql jdbc? I think we can fix this issue without this change.

Comment (eye-gu) [reply dolphinscheduler-bom/pom.xml]: The upgrade is to ensure that the fields in the database are null and the objects are also null, rather than enumerating 0

File: dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/entity/Command.java [hunk 2]
```diff
@@ -61,6 +61,7 @@ public class Command {
     private String commandParam;
 
     @TableField("task_depend_type")
+    @Builder.Default
```
Comment (ruanwenjun) [inline dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/entity/Command.java:64]: Please add `@Builer.Default` to `failureStrategy` and `updateTime`.

In fact I don't think it's a good idea to add default value in DO， but this is a history issue.

**Review by ruanwenjun — CHANGES_REQUESTED** [review]

File: dolphinscheduler-bom/pom.xml (hunk 1 above)
Comment (ruanwenjun) [reply dolphinscheduler-bom/pom.xml]: The enums are parsed by Mybatis-plus, I don't think this should be related to MySQL driver, is there any related doc?

BTW, if there is a reason to upgrade the MySQL driver, it's needed to use a single PR.

Comment (eye-gu) [reply dolphinscheduler-bom/pom.xml]: This bug is caused by both the MySQL driver and mybatisplus. The driver returned 0 when the database field was null, and mybatisplus did not correctly use `wasnull` to determine. You can refer to this link specifically: https://github.com/baomidou/mybatis-plus/issues/5266

Comment (ruanwenjun) [reply dolphinscheduler-bom/pom.xml]: This should be a serious bug, MySQL stores null but the driver returns 0.

I find the bug is `ResultSetImpl.getObject()` will return a non null value. But I havn't test this.
https://dev.mysql.com/doc/relnotes/connector-j/8.0/en/news-8-0-17.html

File: dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/entity/Command.java (hunk 2 above)
Comment (eye-gu) [reply dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/entity/Command.java:64]: Added

File: dolphinscheduler-bom/pom.xml (hunk 1 above)
Comment (ruanwenjun) [inline dolphinscheduler-bom/pom.xml]: Could please split this to another PR?

**Review by ruanwenjun — CHANGES_REQUESTED** [review]

File: dolphinscheduler-bom/pom.xml (hunk 1 above)
Comment (eye-gu) [reply dolphinscheduler-bom/pom.xml]: Okay

**Review by ruanwenjun — APPROVED** [review]
